scripts/Merge_Trajectories.py

```diff
@@ -0,0 +1,190 @@
+            if line != "\n" and not line.startswith("#"):
+                current_frame = FramefromLine(line)
+                if last_frame == not_yet_found:
+                    last_frame = current_frame
+                elif last_frame != current_frame and last_frame + 1 != current_frame:
+                    raise IncorrectTrajectoryException(
+                        f"file {file} has missing frames ~ missed frame {last_frame+1}")
+                else:
+                    last_frame = current_frame
+    return True
+
+
+def startingFrame(file):
+    with open(file, "r") as opened_file:
+        for line in opened_file:
+            if line != "\n" and not line.startswith("#"):
+                return FramefromLine(line)
+
+
+def endingFrame(file):
+    with open(file, "r") as opened_file:
+        lines = opened_file.readlines()
+        lines.reverse()
+
+    for line in lines:
+        if line != "\n" and not line.startswith("#"):
+            return FramefromLine(line)
+
+
+def checkdata(files):
+    """checks if files are matching one Trajectory"""
+    end_frames = []
+    start_frames = []
+    for file in files:
+        if not isComplete(file):
+            return False
+        start_frames.append(startingFrame(file))
+        end_frames.append(endingFrame(file))
+    for start_frame in start_frames:
+        # a starting frame is considered valid if it is the first frame of the given trajectories
+        # a starting frame is considered valid if it comes after an end frame
+        if start_frame != min(start_frames) and start_frame - 1 not in end_frames:
+            raise IncorrectTrajectoryException(f"the starting frame {start_frame} is not matching the other files")
+        # a starting frame is invalid if it appears more than once
+        if start_frames.count(start_frame) > 1:
+            raise IncorrectTrajectoryException(f"multiple files start with the frame {start_frame}")
+    return True
+
+
+def addDatas(trajecs, output, debug):
+    """will merge all data into the output-file
+      "trajecs" is a list of all Trajectory files"""
+    files = trajecs
+
+    # creates a list with all first Frames
+    first_frames = []
+    for file in files:
+        first_frames.append(startingFrame(file))
+
+    debug_counter = 1
+    debug_length = len(first_frames)
+    # until all files are written
+    while first_frames:
+        # determines the lowest starting frame
+        next_frame = min(first_frames)
+        for i, frame in enumerate(first_frames):
+            # adds all data from that file to the output file
+            if frame == next_frame:
+                if debug:
+                    log.debug(f"writing file {debug_counter}/{debug_length}")
+                write(files[i], output)
+                # removes file and the first line from the lists after the data has been written
+                files.pop(i)
+                first_frames.pop(i)
+                debug_counter += 1
+                break
```

Comment:
I understand what you do here but I was expecting to see something like:

1.) Sort files by min frame
2.) Append frames from each file to output file

Instead of having two lists, one for starting frames and another for the filename you can create one list with a tuple of `(filename, min_frame_id)` and then sort this list ascending

E.g.
```Python
file_info = []
for t in trajecs:
    file_info.append((t, startingFrame(t)))
file_info.sort(key=lambda x: x[1])
```

Or more idiomatic with a list comprehension 

```Python
file_info = [(f, startingFrame(f)) for f in trajecs]
file_info.sort(key=lambda x: x[1])
```

Followed by

```Python
for f, _ in file_info:
    write(f, output)
```

Reply:
I have thought about a concept where each file gets a class in which important values saved. These values are calculated in the beginning so that later on they don´t have to be determined again.
Important values may include:
Path
starting & endframe
framerate & geometry

this Concept still needs to be implemented and is not finished

Reply:
Implemented using Trajectory-Class